.github/workflows/required-tests.yml

```diff
@@ -0,0 +1,211 @@
+      container: fedora:${{ matrix.os }}
+      strategy:
+          matrix:
+            os: ['30', '31']
+      steps:
+          - name: Update base image
+            run: |
+                  set -x &&
+                  dnf update -y &&
+                  dnf install -y dnf-plugins-core rpm-build git
+
+          - name: Clone the repository
+            uses: actions/checkout@v2
+
+          - name: Install PKI build deps
+            run: |
+                  # dnf copr enable @pki/master
+                  dnf builddep -y --allowerasing --spec ./pki.spec &&
+                  dnf clean all
+
+          - name: Build PKI packages
+            run: ./build.sh --with-timestamp --with-commit-id --work-dir=../packages rpm
+
+          - name: Compress RPMS
+            run: tar -czf pki-rpms.tar.gz ../packages/RPMS/*
+
+          # upload-artifact runs on host-vm rather than inside the container. Fixed in v2 (unreleased)
+          # Bug: https://github.com/actions/upload-artifact/issues/13#issuecomment-532936650
+          - name: Upload RPM artifacts
+            uses: actions/upload-artifact@v1
+            with:
+              name: pki-build-${{ matrix.os }}
+              path: pki-rpms.tar.gz
+    
+    # Test job
+    pki-tests:
+      # This job depends on the 'build' job and waits till it completes.
+      # This job needs container to be started manually, as Github provided
+      # container **does not** use /usr/bin/init as its ENTRYPOINT.
+      name: Test PKI
+      needs: build
+      runs-on: ubuntu-latest
+      env:
+        CONTAINER: pkitest
+        BUILDDIR: /tmp/workdir
+        PKIDIR: /tmp/workdir/pki
+        LOGS: ${GITHUB_WORKSPACE}/logs.txt
+        COPR_REPO: "@pki/master"
+      strategy:
+        matrix:
+          os: ['30', '31']
+      steps:
+        - name: Clone the repository
+          uses: actions/checkout@v2
+
+        - name: Setup required test environment
+          run: IMAGE=fedora:${{ matrix.os }} travis/runner-init.sh
+
+        - name: Download PKI binaries from Build job
+          uses: actions/download-artifact@v1
+          with:
+            name: pki-build-${{ matrix.os }}
+
+        - name: Extract tar.gz for rpms
+          run: tar -xzf pki-build-${{ matrix.os }}/pki-rpms.tar.gz
+
+        - name: Install required packages
+          run: docker exec -i ${CONTAINER} dnf install -y 389-ds-base-legacy-tools findutils dnf-plugins-core sudo wget 389-ds-base
+
+        - name: Install PKI packages
+          run: docker exec -i ${CONTAINER} bash -c "find ${PKIDIR} -name '*.rpm' -and -not -name '*debuginfo*' | xargs dnf -y install"
+        
+        #- name: Enable PKI COPR repo
+        #  run: docker exec -i ${CONTAINER} dnf copr enable -y ${COPR_REPO}
+
+        - name: Install DS
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/ds-create.sh
+
+        - name: Install CA
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/ca-create.sh
```

Comment:
If the installation fails we have code in `ca-create.sh` to upload the logs to transfer.sh for troubleshooting. Can that code be changed to upload to GH instead eventually?

Reply:
Yes. That's possible and easier too.

Since this patch uploads all the logs in `/var/log/pki`, we can do a minor change in future to `[subsystem]-create.sh`:
`SYSTEMD_LOG=/var/log/pki/pkitest-systemd-[subsystem].log`

FWIW, this patch [includes PKI journalctl logs](https://github.com/dogtagpki/pki/pull/383/files/931d9a919b39debab875d3e8762e2780cd4c6333#diff-6c601b56e1a5d903c104bc5eb027fca8R134) as an artifact